### PR TITLE
improvement(akita-cli): stop exporting createInitialState from index.ts

### PR DESCRIPTION
**Pull request by theorlovsky**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/datorama/akita/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: #421 

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by theorlovsky** [comment]
@NetanelBasal please take a look at the both `yarn.lock` files.

The root one contained versions different from ones from the `package.json`. I ran `yarn install` and it changed `yarn.lock` accordingly. I'm not sure if those are valid changes.

The CLI's one contained errors. For example, a package called `get-_value`, which doesn't exist, it should be `get-value`. I'm not sure how this happened. Anyway, I renamed those invalid packages and ran `yarn install` from the CLI's directory.

Is it ok?

**Comment by theorlovsky** [comment]
Also, I'm not sure if it's a breaking change. What do you think?

File: tools/akita-cli/templates/angular/index.tpl [hunk 1]
```diff
@@ -1,8 +1,8 @@
 export * from './{{dashCase name}}.query';
-export * from './{{dashCase name}}.store';
+export { {{ pascalCase name }}State, {{pascalCase name}}Store } from './{{dashCase name}}.store';
```
Comment (NetanelBasal) [inline tools/akita-cli/templates/angular/index.tpl:2]: Why not omit the export from `createInitialState`?

**Comment by NetanelBasal** [comment]
> The root one contained versions different from ones from the package.json. I ran yarn install and it changed yarn.lock accordingly. I'm not sure if those are valid changes.

Which dependencies don't exist in the akita-cli folder?

**Comment by NetanelBasal** [comment]
> Also, I'm not sure if changing this export is a breaking change. What do you think?

I don't think so. It'll not break an existing code.

**Comment by NetanelBasal** [comment]
Note that you should also update the part of the schematic.

File: tools/akita-cli/templates/angular/index.tpl (hunk 1 above)
Comment (theorlovsky) [reply tools/akita-cli/templates/angular/index.tpl:2]: You might want to use this function in a service. So, it should be exported, but only inside a store. But if you find it a rare use case, I can change it.

Comment (NetanelBasal) [reply tools/akita-cli/templates/angular/index.tpl:2]: Sounds ok.

**Comment by theorlovsky** [comment]
> > The root one contained versions different from ones from the package.json. I ran yarn install and it changed yarn.lock accordingly. I'm not sure if those are valid changes.
> 
> Which dependencies don't exist in the akita-cli folder?

They all have `-_` in their names, while it should be just `-`: `get-_value`, `has-_value`, `set-_value`, `union-_value`, `unset-_value`. Yarn throws an error while trying to install those dependencies.

**Comment by theorlovsky** [comment]
> Note that you should also update the part of the schematic.

Which schematic do you mean? I haven't found any that requires changes, but I might miss something.

**Comment by NetanelBasal** [comment]
@snebjorn do you know something about the dependencies issues?

**Comment by NetanelBasal** [comment]
There is https://github.com/datorama/akita/tree/master/libs/akita-schematics/src/ng-g but I see that only the CLI creates the `index.ts` file. Are you using the CLI?

**Comment by theorlovsky** [comment]
> There is https://github.com/datorama/akita/tree/master/libs/akita-schematics/src/ng-g but I see that only the CLI creates the `index.ts` file. Are you using the CLI?

Yes, I am

**Comment by snebjorn** [comment]
> @snebjorn do you know something about the dependencies issues?

@NetanelBasal I haven't gotten around to checking it, but I think it's related to https://github.com/datorama/akita/commit/31b1b879f8e965c0955a718c9af6c9b90bc16792

**Comment by NetanelBasal** [comment]
Let us know when you can, please.

**Comment by theorlovsky** [comment]
@NetanelBasal I can exclude those files from the PR, anyway they’re not directly related to the desired changes. After that I can open an issue for dependencies.

**Comment by NetanelBasal** [comment]
Sure

**Comment by theorlovsky** [comment]
Done

**Pull request merged**